Compiler/Util/List.mo

```diff
@@ -1044,6 +1045,7 @@ algorithm
 
     arrayUpdate(arr, i, false);
   end for;
+  GC.free(arr);
```

Comment:
Does this improve performance? There are some more places where temporary arrays could be free'd